### PR TITLE
Add Jenkins Infra implementation notes to README

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I made some notes about the structure of this repository. It might be helpful for contributors who try to understand how to update services and edit their configurations. Will appreciate reviews and feedback about feasibility

CC @MarkEWaite @olblak

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,37 @@ This repository is the [r10k](https://github.com/adrienthebo/r10k) control
...
+
+See the [Jenkins infrastructure project](https://jenkins.io/projects/infrastructure/) for overview of the project's infrastructure and the services being managed by this repository.
+Full list of services is available [here](https://jenkins.io/projects/infrastructure/#services).
+
+### Implementation notes
+
+* The services are managed [r10k](https://github.com/adrienthebo/r10k) and Puppet,
+  configuration files are available inside this repository.
+* There are multiple types of service deployments:
+  * The majority of services run as containers inside Kubernetes, and it is a precondition for new services.
```
Comment (oleg-nenashev) [inline README.md:19]: I should lkely add a reference to https://github.com/jenkins-infra/charts , CC @halkeye

File: README.md [hunk 2]
```diff
@@ -6,6 +6,37 @@ This repository is the [r10k](https://github.com/adrienthebo/r10k) control
 repository for the [Jenkins](https://jenkins.io) project's own
 infrastructure.
 
+## Structure
+
+See the [Jenkins infrastructure project](https://jenkins.io/projects/infrastructure/) for overview of the project's infrastructure and the services being managed by this repository.
+Full list of services is available [here](https://jenkins.io/projects/infrastructure/#services).
```
Comment (olblak) [inline README.md]:
```suggestion
A non exhaustive list of services is available [here](https://jenkins.io/projects/infrastructure/#services).
```

**Review by olblak — APPROVED** [review]
On overall it's a great improvement

**Comment by oleg-nenashev** [comment]
@olblak Created https://issues.jenkins-ci.org/browse/WEBSITE-678 as a follow-up to get other services documented

File: README.md (hunk 1 above)
Comment (oleg-nenashev) [reply README.md:19]:
```suggestion
  * The majority of services run as containers inside Kubernetes, and it is a precondition for new services.
    There is ongoing work on [Jenkins Infra Helm charts](https://github.com/jenkins-infra/charts) which are expected to replace puppet deployments in the future.
```

**Review by timja — APPROVED** [review]

**Pull request merged**